browser/brave_content_browser_client.cc

```diff
@@ -1253,9 +1253,11 @@ BraveContentBrowserClient::CreateThrottlesForNavigation(
 #endif
 
 #if BUILDFLAG(ENABLE_AI_CHAT)
-  if (auto ai_chat_throttle =
-          ai_chat::AiChatThrottle::MaybeCreateThrottleFor(handle)) {
-    throttles.push_back(std::move(ai_chat_throttle));
+  if (brave::IsRegularProfile(context)) {
```

Comment:
we shouldn't be making these kinds of checks here, it's very fragile and has to be copied in multiple places like `BraveRenderViewContextMenu::IsAIChatEnabled`. These checks need to be consolidated somewhere.